### PR TITLE
Fix handling of quoted boolean values in `YAML::Any`

### PR DESCRIPTION
**Pull request by willhbr**
`"yes"` and `"no"` should be treated as strings (since they're quoted) whereas `yes` and `no` should be booleans (since they're plain). This is already handled in most places, but `YAML::Any` passes the node value, rather than the node itself when handling a scalar and so the quoting type is lost, and everything is parsed as though it is unquoted.

This is a breaking change.

### CODE REVIEW

File: spec/std/yaml/any_spec.cr [hunk 1]
```diff
@@ -94,11 +94,15 @@ describe YAML::Any do
       YAML.parse("").as_nil.should be_nil
     end
 
-    it "gets bool" do
+    it "gets bool", focus: true do
```
Comment (HertzDevil) [inline spec/std/yaml/any_spec.cr:97]:
```suggestion
    it "gets bool" do
```

**Review by HertzDevil — CHANGES_REQUESTED** [review]

File: spec/std/yaml/any_spec.cr [hunk 2]
```diff
@@ -94,11 +94,15 @@ describe YAML::Any do
...
       YAML.parse("true").as_bool.should be_true
       YAML.parse("false").as_bool.should be_false
       YAML.parse("true").as_bool?.should be_true
       YAML.parse("false").as_bool?.should be_false
+      YAML.parse("yes").as_bool?.should be_true
+      YAML.parse("no").as_bool?.should be_false
+      YAML.parse("'yes'").as_bool?.should be_nil
+      YAML.parse("'no'").as_bool?.should be_nil
```
Comment (HertzDevil) [inline spec/std/yaml/any_spec.cr:105]: These specs pass even without this PR. Instead this PR seems to only affect users of `YAML::Serializable::Unmapped`:

```crystal
require "yaml"

class Foo
  include YAML::Serializable
  include YAML::Serializable::Unmapped
end

# before
Foo.from_yaml("x: yes")      # => #<Foo:0x13732ca0ce0 @yaml_unmapped={"x" => true}>
Foo.from_yaml("x: 'yes'")    # => #<Foo:0x13732ca0bc0 @yaml_unmapped={"x" => true}>
YAML.parse("'yes'").as_bool? # => nil

# after
Foo.from_yaml("x: yes")   # => #<Foo:0x24f0b1d0cc0 @yaml_unmapped={"x" => true}>
Foo.from_yaml("x: 'yes'") # => #<Foo:0x24f0b1d0ba0 @yaml_unmapped={"x" => "yes"}>
YAML.parse("'yes'").as_bool? # => nil
```

Comment (willhbr) [reply spec/std/yaml/any_spec.cr]: Ah `YAML.parse` and `YAML::Any.from_yaml` seem to go through different codepaths?

I can't see any specs specifically for `YAML::Any.from_yaml`—should I add some in this file?

And yeah this will change the parsing of `YAML::Unmapped` or if you have a field of type `YAML::Any` (which is what I was doing).

So:

```
require "yaml"

# Before
YAML::Any.from_yaml "'yes'" # => true
YAML::Any.from_yaml "yes" # => true
YAML.parse "'yes'" # => "yes"
YAML.parse "yes" # => true

# Before
YAML::Any.from_yaml "'yes'" # => "yes"
YAML::Any.from_yaml "yes" # => true
YAML.parse "'yes'" # => "yes"
YAML.parse "yes" # => true
```

File: spec/std/yaml/any_spec.cr (hunk 1 above)
Comment (willhbr) [reply spec/std/yaml/any_spec.cr]: Sorry, fixed now.

File: spec/std/yaml/any_spec.cr (hunk 2 above)
Comment (HertzDevil) [reply spec/std/yaml/any_spec.cr]: Please do add the necessary specs

Comment (straight-shoota) [reply spec/std/yaml/any_spec.cr]: > Ah `YAML.parse` and `YAML::Any.from_yaml` seem to go through different codepaths?

Yeah, that's all a bit weird and hard to follow through, what's actually going on. YAML parsing could probably use some refactoring just to get the code clarified.

**Review by straight-shoota — APPROVED** [review]

File: spec/std/yaml/any_spec.cr (hunk 2 above)
Comment (willhbr) [reply spec/std/yaml/any_spec.cr]: Tests added :)

**Review by HertzDevil — APPROVED** [review]

**Comment by straight-shoota** [comment]
Removed the `breaking` label. This is a bug fix and only breaks code that relied on that pretty obvious bug.

ref: https://forum.crystal-lang.org/t/breaking-changes/5805/2?u=straight-shoota
